app/src/main/java/net/inkyquill/pocketeditor/yandex/YandexDiskGateway.kt

```
@@ -317,6 +336,246 @@ class OkHttpYandexDiskGateway(
+                withContext(NonCancellable) { recoverManifestPublication(rootPath, ownedLock) }
+            }.exceptionOrNull()
+            if (recoveryFailure != null) failure.addSuppressed(recoveryFailure)
+            throw failure
+        }
+    }
+
+    override suspend fun recoverManifestPublication(rootPath: String, ownedLock: SyncLock) {
+        recoverManifestPublication(rootPath, ownedLock, resumableTransactionId = null)
+    }
+
+    private suspend fun recoverManifestPublication(
```

detekt (1.23.8)

[warning] The function recoverManifestPublication appears to be too complex based on Cyclomatic Complexity (complexity: 52). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CyclomaticComplexMethod)